src/scripting/lua_widget_attributes.cpp

```diff
@@ -342,16 +420,328 @@ WIDGET_SETTER("marked_up_text", t_string, gui2::styled_widget)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::rich_label)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("link_aware", bool, gui2::scroll_label)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::scroll_label)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("link_aware", bool, gui2::scroll_text)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::scroll_text)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("link_color", std::string, gui2::label)
+{
+	return w.get_link_color().to_hex_string();
+}
+
+WIDGET_SETTER("link_color", std::string, gui2::label)
+{
+	w.set_link_color(color_t::from_hex_string(value));
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("link_color", std::string, gui2::rich_label)
+{
+	return w.get_link_color().to_hex_string();
+}
+
+WIDGET_SETTER("link_color", std::string, gui2::rich_label)
+{
+	w.set_link_color(color_t::from_hex_string(value));
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("max_input_length", int, gui2::combobox)
+{
+	return w.get_max_input_length();
+}
+
+WIDGET_SETTER("max_input_length", int, gui2::combobox)
+{
+	if(value < 0) {
+		throw std::invalid_argument("max_input_length must be >= 0");
+	}
+	w.set_max_input_length(value);
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("max_input_length", int, gui2::text_box)
+{
+	return w.get_max_input_length();
+}
+
+WIDGET_SETTER("max_input_length", int, gui2::text_box)
+{
+	if(value < 0) {
+		throw std::invalid_argument("max_input_length must be >= 0");
+	}
+	w.set_max_input_length(value);
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("step_size", int, gui2::slider)
+{
+	return w.get_step_size();
+}
+
+WIDGET_SETTER("step_size", int, gui2::slider)
+{
+	if(value < 0) {
+		throw std::invalid_argument("step_size must be >= 0");
+	}
+	w.set_step_size(value);
+}
+
+WIDGET_GETTER("text_alignment", std::string, gui2::styled_widget)
+{
+	std::string s;
+
+	switch(w.get_text_alignment()) {
+		case(PangoAlignment::PANGO_ALIGN_LEFT):
+			s = "left";
+			break;
+		case(PangoAlignment::PANGO_ALIGN_RIGHT):
+			s = "right";
+			break;
+		case(PangoAlignment::PANGO_ALIGN_CENTER):
+			s = "center";
+	}
+
+	return s;
+}
+
+WIDGET_SETTER("text_alignment", std::string, gui2::styled_widget)
+{
+	if(value == "left") {
+		w.set_text_alignment(PangoAlignment::PANGO_ALIGN_LEFT);
+	} else if(value == "right") {
+		w.set_text_alignment(PangoAlignment::PANGO_ALIGN_RIGHT);
+	} else if(value == "center") {
+		w.set_text_alignment(PangoAlignment::PANGO_ALIGN_CENTER);
+	} else {
+		throw std::invalid_argument("text_alignment must be one of <left,center,right>");
+	}
+}
+
+WIDGET_GETTER("tooltip", t_string, gui2::styled_widget)
+{
+	return w.tooltip();
+}
+
 WIDGET_SETTER("tooltip", t_string, gui2::styled_widget)
 {
 	w.set_tooltip(value);
 }
 
+WIDGET_GETTER("use_tooltip_on_label_overflow", bool, gui2::styled_widget)
```

Comment:
This… is a bit long… it's not really a major problem, but if there's a way to make it a bit shorter, that might be nice…

Reply:
overflow_to_tooltip isn't much better, but I'll go with that unless you come up with something.